apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/default-strategies.ts

```diff
@@ -18,7 +14,14 @@ export function getDefaultStrategyCredentialsMap(): StrategyCredentialsMap {
 
 const localCredentials: StrategyCredentials = {
   name: 'local',
-  work: (cdRef) => cdRef.detectChanges(),
+  work: (cdRef, _, notification) => {
+    if(['rxComplete', 'rxError'].includes(notification.kind)) {
+      cdRef.detach();
+    } else {
+      cdRef.reattach();
+      cdRef.detectChanges();
+    }
+  },
```

Comment:
so the error and complete template will not receive any changes? don't they receive some context which should get updated?
Also, I think this implementation is very slow, we should store the methods in a map and just call

```ts
notificationWorkMap[notification.kind]();
```

this way we save 1.000s of iterations over the same array again and again.

Reply:
"cold" was confusing at the beginning for me. But got it. 
You mean values from the observable. 

There is no next value after error or complete that could occur from the observable. 
And the triggers for the template has to behave the same. Detach temple and no other CDs affect it. A potential next value will reattach the view.

For the map:
Sure we have to make to nice! Fir the Poc it was important to see what we need to change in the strategy config.

So I will skip any changes on low level implementations.